addons/misra.py

```diff
@@ -2628,10 +2628,9 @@ def isRuleSuppressed(self, file_path, linenr, ruleNum):
         # Remove any prefix listed in command arguments from the filename.
         filename = None
         if file_path is not None:
-            if self.filePrefix is not None:
-                filename = remove_file_prefix(file_path, self.filePrefix)
```

Comment:
Removing `filePrefix` logic is not a good idea.

I had delete it in one of my PRs, and this broke Cppcheck for some users (see https://github.com/danmar/cppcheck/pull/1996#issuecomment-514180830). So I had to restore it in https://github.com/danmar/cppcheck/pull/2030.

Therefore, we must keep the logic of this file prefix (and `-P` argument for misra.py).

Reply:
However, if necessary, you must inform users that this argument will be removed after several releases of Cppcheck. And add deprecation warnings. But I don't see the need to get rid of file prefix right now.